libsolidity/ast/ASTJsonImporter.cpp

```diff
@@ -0,0 +1,995 @@
+{
+	std::vector<ASTPointer<Expression>> arguments;
+	for (auto& arg: member(_node, "arguments"))
+		arguments.push_back(convertJsonToASTNode<Expression>(arg));
+	return createASTNode<ModifierInvocation>(
+		_node,
+		createIdentifier(member(_node, "modifierName")),
+		member(_node, "arguments").isNull() ? nullptr : make_unique<std::vector<ASTPointer<Expression>>>(arguments)
+	);
+}
+
+ASTPointer<EventDefinition> ASTJsonImporter::createEventDefinition(Json::Value const&  _node)
+{
+	return createASTNode<EventDefinition>(
+		_node,
+		memberAsASTString(_node, "name"),
+		nullOrASTString(_node, "documentation"),
+		createParameterList(member(_node, "parameters")),
+		memberAsBool(_node, "anonymous")
+	);
+}
+
+ASTPointer<ElementaryTypeName> ASTJsonImporter::createElementaryTypeName(Json::Value const& _node)
+{
+	unsigned short firstNum;
+	unsigned short secondNum;
+
+	astAssert(_node["name"].isString(), "Expected 'name' to be a string!");
+
+	string name = member(_node, "name").asString();
+	Token token;
+	tie(token, firstNum, secondNum) = TokenTraits::fromIdentifierOrKeyword(name);
+	ElementaryTypeNameToken elem(token, firstNum,  secondNum);
+
+	std::optional<StateMutability> mutability = {};
+	if (_node.isMember("stateMutability"))
+		mutability = stateMutability(_node);
+
+	return createASTNode<ElementaryTypeName>(_node, elem, mutability);
+}
+
+ASTPointer<UserDefinedTypeName> ASTJsonImporter::createUserDefinedTypeName(Json::Value const& _node)
+{
+	astAssert(_node["name"].isString(), "Expected 'name' to be a string!");
+
+	vector<ASTString> namePath;
+	vector<string> strs;
+	string nameString = member(_node, "name").asString();
+	boost::algorithm::split(strs, nameString, boost::is_any_of("."));
+	for (string s: strs)
+		namePath.push_back(ASTString(s));
+	return createASTNode<UserDefinedTypeName>(
+		_node,
+		namePath
+	);
+}
+
+ASTPointer<FunctionTypeName> ASTJsonImporter::createFunctionTypeName(Json::Value const&  _node)
+{
+	return createASTNode<FunctionTypeName>(
+		_node,
+		createParameterList(member(_node, "parameterTypes")),
+		createParameterList(member(_node, "returnParameterTypes")),
+		visibility(_node),
+		stateMutability(_node)
+	);
+}
+
+ASTPointer<Mapping> ASTJsonImporter::createMapping(Json::Value const&  _node)
+{
+	return createASTNode<Mapping>(
+		_node,
+		createElementaryTypeName(member(_node, "keyType")),
+		convertJsonToASTNode<TypeName>(member(_node, "valueType"))
+	);
+}
+
+ASTPointer<ArrayTypeName> ASTJsonImporter::createArrayTypeName(Json::Value const&  _node)
+{
+	return createASTNode<ArrayTypeName>(
+		_node,
+		convertJsonToASTNode<TypeName>(member(_node, "baseType")),
+		nullOrCast<Expression>(member(_node, "length"))
+	);
+}
+
+ASTPointer<InlineAssembly> ASTJsonImporter::createInlineAssembly(Json::Value const& _node)
+{
+	yul::Dialect const& dialect = yul::EVMDialect::strictAssemblyForEVM(langutil::EVMVersion());
```

Comment:
I would say this is still a blocker.

Reply:
We should export the dialect.

Reply:
If I recall correctly, we talked about that before in a meeting and there were reasons I don't recall now that we decided against that

Reply:
I think one reason it wasn't that easy was because there was no simple definition of a dialect as it is made of different classes with varying attributes and settings. Related to this is https://github.com/ethereum/solidity/issues/7830

Reply:
Solidity inline assembly has clearly defined dialects. You only need the evm version for that. The dialect is part of the assembly ast node, so we only have to print it. My proposal would be `EVM-petersburg` and so on. We could add a version that we increment when we make changes outside of just the EVM version.

Another solution would be using the EVM version configured from the compiler settings. If we take the first approach, we should also check that it matches the configured version.